lrtable/src/lib/statetable.rs

```diff
@@ -302,10 +307,35 @@ where
         })
     }
 
+    fn decode(bits: u32) -> Action<StorageT> {
+        let action = (bits & 0b11) as u8;
+        let val: u32 = bits >> 2;
+
+        match action {
+            SHIFT => Action::Shift(StIdx::from(val)),
+            REDUCE => Action::Reduce(PIdx(StorageT::from(val).unwrap())),
```

Comment:
I think this can be `PIdx(val.as_())` because we know, by definition, that `val` will fit in `StorageT`. [Using the `as_` should be quite a bit faster than `from(...).unwrap()`.] We need to document *why* the `as_` call is safe though. Something like "Since val was originally stored in a StorageT, we know by definition that it can fit into a StorageT now and hence the call to as_ is safe."

Reply:
Doesn't seem to work: `the trait num_traits::AsPrimitive<StorageT> is not implemented for u32`